src/datarobotx/idp/common/credentials_hooks.py

```diff
@@ -35,17 +40,52 @@ def after_catalog_created(
         load_versions: Dict[str, str],
     ) -> None:
         """Validate credentials and add them to the catalog for node consumption."""
-        self.validate_credentials(conf_creds)
+        self._validate_or_replace_credentials(conf_creds)
 
-        creds = get_feed_dict(conf_creds, copy_dict_as=None, key_prefix="params:credentials.")
+        creds = get_feed_dict(
+            conf_creds, copy_dict_as=None, key_prefix="params:credentials."
+        )
         catalog.add_feed_dict(creds)
 
-    @staticmethod
-    def validate_credentials(conf_creds: Dict[str, Any]) -> None:
+    def _validate_or_replace_credentials(self, conf_creds: Dict[str, Any]) -> None:
         """Confirm credentials appear to be valid and functional."""
         import datarobot as dr
 
-        dr_creds = conf_creds["datarobot"]
+        exception_occurred = False
+        try:
+            dr_creds = conf_creds["datarobot"]
+            dr.Client(token=dr_creds["api_token"], endpoint=dr_creds["endpoint"])  # type: ignore
+        except ValueError:  # Invalid or unpopulated credentials
+            exception_occurred = True
+        except KeyError:  # No credentials.yml file
+            exception_occurred = True
+            conf_creds["datarobot"] = {
+                "prediction_environment_id": None,
+            }
+
+        finally:
```

Comment:
Noting that I can delete this line, but I never get the chance to write `finally` after try/except